.github/workflows/test.yml

```diff
@@ -219,9 +219,11 @@ jobs:
 
       - name: Set variables
         id: vars
+        # GITHUB_HEAD_REF is only defined in PRs. GITHUB_REF is on the default branch something like refs/heads/8.x-4.x.
+        # So if GITHUB_HEAD_REF is not defined we use GITHUB_REF and remove leading refs/heads/. On the result / will replaced by _
         run: |
-          echo ::set-output name=DOCKER_SANITIZED_BRANCH_NAME::${GITHUB_HEAD_REF/\//_}
-          echo ::set-output name=DOCKER_TAG::${GITHUB_HEAD_REF/\//_}-${{ github.run_id }}
+          echo ::set-output name=DOCKER_SANITIZED_BRANCH_NAME::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}} | sed 's/\//_/g')
+          echo ::set-output name=DOCKER_TAG::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}} | sed 's/\//_/g')-${{ github.run_id }}
```

Comment:
If we are fine with just having the part of the branch, that is after the last "/" we could simplify like this:

```
echo ::set-output name=DOCKER_SANITIZED_BRANCH_NAME::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF##*/}})
echo ::set-output name=DOCKER_TAG::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF##*/}})-${{ github.run_id }}

```